### PR TITLE
MAINT: Refactor the code used to compute sha256, md5 hashes

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by eric-wieser** [comment]
LGTM. @charris, is this vendorized code we shouldn't be changing?

**Comment by eric-wieser** [comment]
I'm talking garbage, we can definitely clean this up. Commit message should start with MAINT:, but otherwise ready to merge. This will conflict with your other PR, so you're making work for yourself here!

**Comment by srinivasreddy** [comment]
Not a problem. I will rebase. :smiley:

File: pavement.py [hunk 1]
```diff
@@ -562,25 +558,21 @@ def sdist(options):
         target = os.path.join(idirs, tarball_name(t))
         shutil.copy(source, target)
 
-def compute_md5(idirs):
+def _compute_hash(idirs, algo):
     released = paver.path.path(idirs).listdir()
     checksums = []
     for f in sorted(released):
-        m = md5(open(f, 'r').read())
+        m = algo(open(f, 'r').read())
```
Comment (eric-wieser) [inline pavement.py]: While you're here, mind making a second commit that uses `with` to correctly close the file?

**Comment by srinivasreddy** [comment]
@eric-wieser  Not sure what's happening with appveyor . Could you please look into it?

**Comment by charris** [comment]
@eric-wieser Most of the pavement file is outdated, I'm going to remove a lot of it sometime in the future.

**Comment by charris** [comment]
Thanks @srinivasreddy .

**Pull request merged**